dask/bag/core.py

```diff
@@ -1772,12 +1772,15 @@ def from_sequence(seq, partition_size=None, npartitions=None):
     """
     seq = list(seq)
     if npartitions and not partition_size:
-        partition_size = int(math.ceil(len(seq) / npartitions))
+        if len(seq) <= 100:
+            partition_size = int(math.ceil(len(seq) / npartitions))
+        else:
+            partition_size = max(1, int(math.floor(len(seq) / npartitions)))
```

Comment:
I tweaked this so that the new heuristic is only applied on bags with more than 100 items, similar to automatic partitioning. We could remove this and just use floor/max for everything but a number of tests would need to be changed as it would break some assumptions we have with small bags today.